### PR TITLE
Reduce pre flight mag inconsistency test false positives

### PR DESCRIPTION
**Pull request by priseborough**
Work in progress for comment and review.

Raised in response to https://github.com/PX4/Firmware/issues/12272.

@MaEtUgR are you able to test on your setup(s) and provide feedback? TXS

### CODE REVIEW

**Comment by dagar** [comment]
Under what circumstances would you advise relaxing the thresholds as an acceptable solution?

I guess what I'm really wondering is if we want to expose these tuning knobs versus identifying a mag that might as well be disabled entirely or a switch to skip the check entirely (what most users are actually doing when "tuning" the parameter).

File: src/modules/commander/commander_params.c [hunk 1]
```diff
@@ -599,16 +599,28 @@ PARAM_DEFINE_FLOAT(COM_ARM_IMU_ACC, 0.7f);
...
+ *
+ * @group Commander
+ * @unit rad
+ * @min 0.1
+ * @max 4.0
  * @decimal 2
  * @increment 0.05
  */
-PARAM_DEFINE_FLOAT(COM_ARM_MAG, 0.15f);
+PARAM_DEFINE_FLOAT(COM_ARM_MAG_V, 0.7f);
```
Comment (MaEtUgR) [inline src/modules/commander/commander_params.c]: Why does there need to be a different angle for inclination and declination? As far as I understood the check is to find out if the mag is mounted the wrong way or the sensor orientation is set wrong. Not?

File: src/modules/sensors/voted_sensors_update.cpp [hunk 2]
```diff
@@ -1218,21 +1219,50 @@ void VotedSensorsUpdate::calc_mag_inconsistency(sensor_preflight_s &preflt)
...
+			float mag_down_alternate = gravity_vec.dot(mag_vec_alternate);
+
+			// calculate the vertical angle between the two vectors
+			float mag_vang_selected = atan2f(mag_down_selected, mag_hvec_selected.norm());
+			float mag_vang_alternate = atan2f(mag_down_alternate, mag_hvec_alternate.norm());
+			float ang_v_delta = fabsf(mag_vang_selected - mag_vang_alternate);
+
+			// calculate the horizontal angle between the two vectors to estimate the difference in magnetic heading thee sensors would produce
+			matrix::Quatf q_delta(mag_hvec_selected, mag_hvec_alternate);
+			matrix::Vector3f ang_hvec_delta = q_delta.to_axis_angle();
```
Comment (MaEtUgR) [inline src/modules/sensors/voted_sensors_update.cpp]: `to_axis_angle()` is "XXX DEPRECATED"

File: src/modules/sensors/voted_sensors_update.cpp [hunk 3]
```diff
@@ -1249,10 +1279,12 @@ void VotedSensorsUpdate::calc_mag_inconsistency(sensor_preflight_s &preflt)
 
 	// skip check if less than 2 sensors
 	if (check_index < 1) {
-		preflt.mag_inconsistency_ga = 0.0f;
+		preflt.mag_inconsistency_h = 0.0f;
+		preflt.mag_inconsistency_v = 0.0f;
 
 	} else {
 		// get the vector length of the largest difference and write to the combined sensor struct
```
Comment (MaEtUgR) [inline src/modules/sensors/voted_sensors_update.cpp]: comment outdated

**Comment by MaEtUgR** [comment]
Here's my suggestion that I also successfully tested now: https://github.com/PX4/Firmware/pull/12334

**Pull request closed without merging**